src/porepy/numerics/ad/operator_functions.py

```diff
@@ -170,188 +238,117 @@ def get_values(self, *args: AdArray) -> np.ndarray:
+                passed to the call to this instance.
 
         Returns:
-            Numeric representation of the Jacobian of this function.
+            Function derivatives in numerical format.
 
         """
         pass
 
 
-class AbstractJacobianFunction(AbstractFunction):
-    """Partially abstract base class, providing a call to the callable ``func`` in order to
-    obtain numeric function values.
+class DiagonalJacobianFunction(AbstractFunction):
+    """Partially abstract operator function, which approximates the Jacobian of the
+    function using identities and scalar multipliers per dependency.
 
-    What remains abstract is the Jacobian.
+    Can be used to for functions with approximated derivatives.
 
-    """
-
-    def get_values(self, *args: AdArray) -> np.ndarray:
-        """
-        Returns:
-            The direct evaluation of the callable using ``val`` of passed AD arrays.
+    Parameters:
+        multipliers: Scalar multipliers for the identity blocks in the Jacobian,
+            per function argument. The order in ``multipliers`` is expected to match
+            the order of AD operators passed to the call of this instance.
 
-        """
-        # get values of argument AdArrays.
-        vals = (arg.val for arg in args)
+    """
 
-        # if the callable is flagged as conform for vector operations, feed vectors
-        if self.array_compatible:
-            return self.func(*vals)
+    def __init__(
+        self,
+        multipliers: float | list[float],
+        name: str,
+    ):
+        super().__init__(name=name)
+        # check and format input for further use
+        if isinstance(multipliers, list):
+            self._multipliers = [float(val) for val in multipliers]
         else:
-            # if not vector-conform, feed element-wise
-
-            # TODO this displays some special behavior when val-arrays have different lengths:
-            # it returns None-like things for every iteration more then shortest length
-            # These Nones are ignored for some reason by the function call, as well as by the
-            # array constructor.
-            # If a mortar var and a subdomain var are given as args,
-            # then the lengths will be different for example.
-            return np.array([self.func(*vals_i) for vals_i in zip(*vals)])
-
+            self._multipliers = [float(multipliers)]
 
-### CONCRETE IMPLEMENTATIONS ------------------------------------------------------------------
+    def get_jacobian(self, *args: float | np.ndarray | AdArray) -> sps.spmatrix:
+        """The approximate Jacobian consists of identity blocks times scalar multiplier
+        per every function dependency."""
+        jacs = [
+            arg.jac * m
+            for arg, m in zip(args, self._multipliers)
+            if isinstance(arg, AdArray)
+        ]
+        return sum(jacs).tocsr()
 
 
 class Function(AbstractFunction):
-    """Ad representation of an analytically given function,
```

Comment:
The Function class receives the callable as before, but it internally uses it in the new, inherited `func` member